odoo/addons/test_new_api/models/test_new_api.py

```diff
@@ -156,7 +156,7 @@ class Message(models.Model):
     @api.constrains('author', 'discussion')
     def _check_author(self):
         for message in self.with_context(active_test=False):
-            if message.discussion and message.author not in message.discussion.participants:
+            if message.discussion and message.author not in message.discussion.sudo().participants:
```

Comment:
Since I'm forbidding the user to read the model `test_new_api.discussion`, I got an access error here when the system creates the record and checks that constraint.

I can not use a `sudo` call on the `create` as the issue occurs specifically when the user does not have access to the parent model.

So, I put the `sudo` call here. Do you think it's okay? What else could I do?